src/npmPackage.js

```diff
@@ -161,6 +208,91 @@ export default function npmPackageFactory(rootDir = '', quiet = true) {
             await writePkg(folderName, { ...packageJson, version });
 
             return this.npmCommand('i');
+        },
+
+        /**
+          * For monorepo,
+          * update version in package.json and package-lock.json,
+          * in the root and in the packages.
+          * @param {Object[]} packagesInfo - `[{ packageName: string, packagePath: string, version: string }]`
+          * @param {String} rootVersion - new root package.json version
+          * @return {Promise}
+          */
+        async lernaUpdateVersions(packagesInfo, rootVersion) {
+            //root version
+            // eslint-disable-next-line no-unused-vars
+            const { readme, _id, ...packageJson } = await readPkg({ cwd: rootDir });
+            packageJson.version = rootVersion;
+            await writePkg(rootDir, packageJson);
+
+            //monorepo package versions
+            //note: alternatively:
+            //  1) from the same commit as last release tag, create local tags in format `${packageInfo.packageName}@${packageInfo.lastVersion}`
+            //      For example `@oat-sa-private/ui-core@5.19.0`.
+            //  2) `npx lerna version --no-git-tag-version --yes --conventional-commits --changelog-preset @oat-sa/conventional-changelog-tao --no-changelog`
+            //      It will calulate conventional-commit bump for each package,
+            //      and update all needed things in package.json/package-lock.json files.
+            //      `@oat-sa/conventional-changelog-tao` preset should be installed in the repo root, or otherwise can reference module from tao-extension-release:
+            //      `path.join(path.dirname(fileURLToPath(import.meta.url)),'../node_modules/@oat-sa/conventional-changelog-tao/index.js')`
+            //  3) delete local tags
+            for (const packageInfo of packagesInfo) {
+                // eslint-disable-next-line no-unused-vars
+                const { readme, _id, ...packageJson } = await readPkg({ cwd: packageInfo.packagePath });
+                packageJson.version = packageInfo.version;
+                for (const depPackageInfo of packagesInfo) {
+                    for (const depListKey of ['dependencies', 'devDependencies']) {
+                        if (packageJson[depListKey] && packageJson[depListKey][depPackageInfo.packageName]) {
+                            packageJson[depListKey][depPackageInfo.packageName] = `^${depPackageInfo.version}`;
+                        }
+                    }
+                }
+                await writePkg(packageInfo.packagePath, packageJson);
+
+                const packageLockJson = await this.readPackageLock(packageInfo.packagePath);
+                packageLockJson.version = packageInfo.version;
+                if ((packageLockJson.packages || [])['']?.name === packageInfo.packageName) {
```

Comment:
What is the meaning of the `['']` empty string property? Does `readPkg` just return it like that?

Reply:
Yes, in package-lock v3 it contains info about package itself:

![Screenshot_22](https://github.com/oat-sa/tao-extension-release/assets/38751932/315f868d-4804-4dd9-aaea-78cf42b6ad78)